src/goto-programs/goto_convert.cpp

```diff
@@ -200,8 +194,9 @@ void goto_convertt::finish_computed_gotos(goto_programt &goto_program)
     goto_programt::instructiont &i=*g_it;
     exprt destination=i.code.op0();
 
-    assert(destination.id()==ID_dereference);
-    assert(destination.operands().size()==1);
+    DATA_INVARIANT(
+      destination.id() == ID_dereference, "dereference ID not allowed here");
+    DATA_INVARIANT(destination.operands().size() == 1, "expected 1 argument");
```

Comment:
Again, the important question here isn't _that_ you expect one argument (that much can be seen from the condition) but _why_ it is ok to expect one argument here.

Reply:
I know the message is obvious but the check is also obvious. The dereference expects 1 argument. There's no secret message.